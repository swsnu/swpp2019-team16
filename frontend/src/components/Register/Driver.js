import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Box from '../common/Box/Box';
import Button from '../common/Button/Button';
import Typography from '@material-ui/core/Typography';

const DriverBlock = styled.div``;

Driver.propTypes = {
  form: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
};

function Driver({ form, onChange, onClick }) {
  return (
    <DriverBlock>
      <Box>
        <Typography variant="h4">Email</Typography>
        <input
          type="text"
          id="email-input"
          name="email"
          onChange={onChange}
          value={form.email}
        />
        <Typography variant="h4">Password</Typography>
        <input
          type="password"
          id="password-input"
          name="password"
          onChange={onChange}
          value={form.password}
        />
        <Typography variant="h4">Password Confirm</Typography>
        <input
          type="password"
          id="password-confirmation-input"
          name="passwordConfirmation"
          onChange={onChange}
          value={form.passwordConfirmation}
        />
        <Typography variant="h4">Vehicle Info</Typography>
        <Typography variant="h5">Car Type</Typography>
        <input
          type="text"
          id="car-type-input"
          name="carType"
          onChange={onChange}
          value={form.vehicleInfo}
        />
        <Typography variant="h5">Plate No.</Typography>
        <input
          type="text"
          id="plate-no-input"
          name="plateNo"
          onChange={onChange}
          value={form.vehicleInfo}
        />
        <Box>
          <Button
            children="Confirm"
            onClick={onClick}
            variant="contained"
            fullwidth="false"
          />
        </Box>
      </Box>
    </DriverBlock>
  );
}

export default Driver;
