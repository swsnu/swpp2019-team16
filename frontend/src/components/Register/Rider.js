import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/styles';
import theme from '../../lib/styles/theme';

const useStyles = makeStyles({
  container: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
  },
  whiteBox: {
    boxShadow: '0 0 8px rgba(0, 0, 0, 0.25)',
    padding: '2rem',
    width: '360px',
    background: 'white',
    borderRadius: '2px',
  },
  button: {
    background: theme.palette.primary.dark,
    borderRadius: 5,
    color: 'white',
    cursor: 'pointer',
    paddingLeft: '1rem',
    paddingRight: '1rem',
    paddingBottom: '0.3rem',
    paddingTop: '0.3rem',
    marginBottom: '10px',
    marginTop: '10px',
  },
});

Rider.propTypes = {
  form: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
};

function Rider({ form, onChange, onClick }) {
  const styles = useStyles();
  return (
    <div className={styles.whiteBox}>
      <div className={styles.container}>
        <Typography variant="h4">Rider Info</Typography>
        <Typography variant="h5">Email</Typography>
        <input
          type="text"
          id="email-input"
          name="email"
          onChange={onChange}
          value={form.email}
        />
        <Typography variant="h5">Password</Typography>
        <input
          type="password"
          id="password-input"
          name="password"
          onChange={onChange}
          value={form.password}
        />
        <Typography variant="h5">Password Confirm</Typography>
        <input
          type="password"
          id="password-confirmation-input"
          name="passwordConfirmation"
          onChange={onChange}
          value={form.passwordConfirmation}
        />
        <div
          className={styles.button}
          children="Confirm"
          onClick={onClick}
          variant="contained"
          fullwidth="false"
        />
      </div>
    </div>
  );
}

export default Rider;
