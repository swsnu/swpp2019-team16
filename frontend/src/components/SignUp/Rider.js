import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Box from '../common/Box/Box';
import Button from '../common/Button/Button';
import Typography from '@material-ui/core/Typography';

const RiderBlock = styled.div``;

Rider.propTypes = {
  form: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
};

function Rider({ form, onChange, onClick }) {
  return (
    <RiderBlock>
      <Box>
        <Typography variant="h4">Email</Typography>
        <input
          type="text"
          id="email-input"
          name="email"
          onChange={onChange}
          value={form.email}
        />
        <Typography variant="h4">Password</Typography>
        <input
          type="password"
          id="password-input"
          name="password"
          onChange={onChange}
          value={form.password}
        />
        <Typography variant="h4">Password Confirm</Typography>
        <input
          type="password"
          id="password-confirmation-input"
          name="passwordConfirmation"
          onChange={onChange}
          value={form.passwordConfirmation}
        />
        <Box>
          <Button
            children="Confirm"
            onClick={onClick}
            variant="contained"
            fullwidth="false"
          />
        </Box>
      </Box>
    </RiderBlock>
  );
}

export default Rider;
