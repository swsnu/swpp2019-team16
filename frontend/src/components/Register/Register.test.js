import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Rider from './Rider';
import Driver from './Driver';

describe('<Register />', () => {
  const registerMode = {
    rider: 'rider',
    driver: 'driver',
  };

  it('SHOULD match with snapshot WHEN rider', async () => {
    const form = { email: 'dkim94', password: '123' };
    const mode = registerMode.rider;

    const { container } = render(
      <Rider form={form} onChange={() => {}} onClick={() => {}} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('SHOULD match with snapshot WHEN driver', async () => {
    const form = {
      email: 'dkim94',
      password: '123',
      carType: 'BMW',
      plateNo: '123',
    };
    const mode = registerMode.driver;

    const { container } = render(
      <Driver form={form} onChange={() => {}} onClick={() => {}} />,
    );
    expect(container).toMatchSnapshot();
  });
});
