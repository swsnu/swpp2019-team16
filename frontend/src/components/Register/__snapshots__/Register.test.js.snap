// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> SHOULD match with snapshot WHEN driver 1`] = `
<div>
  <div
    class="sc-bwzfXH uTxCW"
  >
    <div
      class="MuiBox-root MuiBox-root-82"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        Email
      </h4>
      <input
        id="email-input"
        name="email"
        type="text"
        value="dkim94"
      />
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        Password
      </h4>
      <input
        id="password-input"
        name="password"
        type="password"
        value="123"
      />
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        Password Confirm
      </h4>
      <input
        id="password-confirmation-input"
        name="passwordConfirmation"
        type="password"
        value=""
      />
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        Vehicle Info
      </h4>
      <h5
        class="MuiTypography-root MuiTypography-h5"
      >
        Car Type
      </h5>
      <input
        id="car-type-input"
        name="carType"
        type="text"
        value=""
      />
      <h5
        class="MuiTypography-root MuiTypography-h5"
      >
        Plate No.
      </h5>
      <input
        id="plate-no-input"
        name="plateNo"
        type="text"
        value=""
      />
      <div
        class="MuiBox-root MuiBox-root-113"
      >
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-root-114 makeStyles-root-117 MuiButton-contained makeStyles-contained-115 makeStyles-contained-118 MuiButton-contained makeStyles-contained-115 makeStyles-contained-118"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Confirm
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Register /> SHOULD match with snapshot WHEN rider 1`] = `
<div>
  <div
    class="sc-bdVaJa bDWFJH"
  >
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        Email
      </h4>
      <input
        id="email-input"
        name="email"
        type="text"
        value="dkim94"
      />
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        Password
      </h4>
      <input
        id="password-input"
        name="password"
        type="password"
        value="123"
      />
      <h4
        class="MuiTypography-root MuiTypography-h4"
      >
        Password Confirm
      </h4>
      <input
        id="password-confirmation-input"
        name="passwordConfirmation"
        type="password"
        value=""
      />
      <div
        class="MuiBox-root MuiBox-root-33"
      >
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-root-34 makeStyles-root-37 MuiButton-contained makeStyles-contained-35 makeStyles-contained-38 MuiButton-contained makeStyles-contained-35 makeStyles-contained-38"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Confirm
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
